### PR TITLE
new(userspace/libsinsp): check usage of wrong-sized variadic arguments in sinsp tests

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Levearaging c++ variadic templates, ensure that variadic arguments are properly sized (at least for integer-like types, that are the most common mistaken ones).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/base64.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
```
Comment (FedeDP) [inline userspace/libsinsp/base64.h:23]: This fixes the build on my machine.

File: userspace/libsinsp/test/sinsp_with_test_input.h [hunk 2]
```diff
@@ -43,9 +43,104 @@ class sinsp_with_test_input : public ::testing::Test
 	sinsp m_inspector;
 
 	void open_inspector(sinsp_mode_t mode = SINSP_MODE_TEST);
-	scap_evt* add_event(uint64_t ts, uint64_t tid, ppm_event_code, uint32_t n, ...);
+
+	template <class ... Ts>
+	void _check_event_params(ppm_event_code event_type, uint32_t n, Ts && ... inputs)
```
Comment (FedeDP) [inline userspace/libsinsp/test/sinsp_with_test_input.h:48]: The function checks:
* that each integer-like argument is properly sized
* that `n` is actually the number of arguments passed

File: userspace/libsinsp/test/sinsp_with_test_input.h [hunk 3]
```diff
@@ -43,9 +43,104 @@ class sinsp_with_test_input : public ::testing::Test
...
+			 i++;
+		 } (), ...);
+		if (i != n)
+		{
+			throw std::runtime_error("wrong number of arguments: specified " +
+						 std::to_string(n) + " but passed: " + std::to_string(i));
+		}
+	}
+
+#define add_event(ts, tid, code, n, ...) \
```
Comment (FedeDP) [inline userspace/libsinsp/test/sinsp_with_test_input.h:132]: Macro magic to automatically call `check_event_params` for us.

File: userspace/libsinsp/test/sinsp_with_test_input.h [hunk 4]
```diff
@@ -43,9 +43,104 @@ class sinsp_with_test_input : public ::testing::Test
...
+        			_add_event(ts, tid, code, n, ##__VA_ARGS__); \
+ 				_check_event_params(code, n, ##__VA_ARGS__)
+
+	scap_evt* _add_event(uint64_t ts, uint64_t tid, ppm_event_code, uint32_t n, ...);
 	sinsp_evt* advance_ts_get_event(uint64_t ts);
-	sinsp_evt* add_event_advance_ts(uint64_t ts, uint64_t tid, ppm_event_code, uint32_t n, ...);
+
+#define add_event_advance_ts(ts, tid, code, n, ...) \
```
Comment (FedeDP) [inline userspace/libsinsp/test/sinsp_with_test_input.h:139]: Same as above

**Comment by FedeDP** [comment]
Example failure:
```
[ RUN      ] sinsp_with_test_input.EVT_FILTER_cmd_str
unknown file: Failure
C++ exception with description "wrong sized argument 0 passed; expected: 8B, received: 4B" thrown in the test body.
[  FAILED  ] sinsp_with_test_input.EVT_FILTER_cmd_str (0 ms)
```
Will fix failures in this PR, and remove `wip` once is ready.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by LucaGuerra — APPROVED** [review]

**Pull request merged**